### ISSUE TITLE
Document spline node requirements for RQSpline

### ISSUE THREAD

**Issue by oschulz** [bat/MonotonicSplines.jl#26]
With

```julia
using MonotonicSplines, InverseFunctions, Plots

X = [-3.6, -3.51, -3.37, -3.23, -3.04, -2.83, -2.57, -2.33, -2.1, -1.87, -1.65, -1.44, -1.24, -1.03, -0.83, -0.63, -0.46, -0.29, -0.12, 0.04, 0.2, 0.35, 0.49, 0.62, 0.75, 0.88, 0.99, 1.11, 1.22, 1.33, 1.43, 1.53, 1.62, 1.72, 1.81, 1.9, 1.99, 2.08, 2.17, 2.26];
Y = [-3.09, -2.93, -2.77, -2.61, -2.46, -2.3, -2.14, -1.98, -1.82, -1.66, -1.51, -1.35, -1.19, -1.03, -0.87, -0.71, -0.55, -0.4, -0.24, -0.08, 0.08, 0.24, 0.4, 0.55, 0.71, 0.87, 1.03, 1.19, 1.35, 1.51, 1.66, 1.82, 1.98, 2.14, 2.3, 2.46, 2.61, 2.77, 2.93, 3.09];

dYdX = MonotonicSplines.estimate_dYdX(X, Y);

f = RQSpline(X, Y, dYdX)
```

(fairly smooth spline parameters) we get

```julia
julia> inverse(f)(-3.6)
ERROR: DomainError with -0.4346471879176269:
sqrt was called with a negative real argument but will only return a complex result if called with a complex argument. Try sqrt(Complex(x)).
```

and the spline is not monotonic and has discontinuities:

```julia
plot(-6:0.01:6, f)
```

![spline](https://github.com/user-attachments/assets/99a49fa4-a0f6-4a31-af86-483ac2395a8b)

**Comment by Micki-D** [comment]
I believe this issue arises due to the implicit requirements to the spline parameters under the hood: 
    - edge knots at the beginning and end of the interval must sit on the identity line 
    - the derivatives at the edge knots must be set to 1

When I modify your above example to comply with these requirements, I get the following spline: 
```
X_new = deepcopy(X)
Y_new = deepcopy(Y)
dYdX_new = deepcopy(dYdX)

pushfirst!(X_new, -3.8)
pushfirst!(Y_new, -3.8)

push!(X_new, 3.4)
push!(Y_new, 3.4)

pushfirst!(dYdX_new, 1.)
push!(dYdX_new, 1.)

f_new = RQSpline(X_new, Y_new, dYdX_new)

plot(-6:0.01:6, f_new, label = "New spline")
scatter!(X,Y, label = "Previous knots")
scatter!([-3.8, 3.4], [-3.8, 3.4], color = :green, label = "Added edge knots on identity")
```

![modified_spline](https://github.com/user-attachments/assets/d267f57a-c663-469f-8eec-79c1851a96d2)

And the spline behaves correctly: 
```
julia> f_new(-3.6)
-3.09

julia> f_new(-3.8)
-3.8
```

**Comment by Micki-D** [comment]
So to fix this issue, I suggest we add an inbuilt mechanism that guarantees these requirements to the spline parameters are met. 

Perhaps in `rqs_forward()` and `rqs_inverse()`.

**Comment by oschulz** [comment]
Good point - for now, let's just add this to the docs of `RQSpline`.

**Comment by Micki-D** [comment]
Alright!

**Comment by mmikhasenko** [comment]
Just discovered that you guys coded the RQS with batch dimension that I wished for in `Bijectors.jl`.
Nice effort, I think it's going to play well with `Lux.jl`

Concerning the issue, people use `softmax` to convert bin widths and heights to knot locations, and `softplus` for derivative.

E.g. 
https://github.com/mmikhasenko/Bijectors.jl/blob/RQS-refactor/src/bijectors/distrax_rational_quadratic_spline.jl#L56-L61

I think one has to do it to convert an output of NN to a RQS

**Comment by oschulz** [comment]
> Just discovered that you guys coded the RQS with batch dimension that I wished for in Bijectors.jl.
Nice effort, I think it's going to play well with Lux.jl

I hope so! We may have to look into some performance aspects though (e.g. dimension order / memory access of the arrays), though - feedback and contributions more than welcome!

**Comment by oschulz** [comment]
> I think one has to do it to convert an output of NN to a RQS

Yes, we kept that separate on purpose. One might also so a simplex/Dirichlet trafo or so, so we thought it's best left to the user how to transform NN-output and similar to spline parameters.

**Comment by mmikhasenko** [comment]
good.
I understood that this issue is about `estimate_dYdX` - I agree that it does not need to know about `softmax` and `softplus`
